resources/views/admin/layouts/master.blade.php

```diff
@@ -33,6 +33,70 @@
     <script src="{{ asset('bower_components/highcharts/highcharts.js') }}"></script>
     <script src="{{ asset('bower_components/highcharts/modules/exporting.js') }}"></script>
     <script src="{{ asset('bower_components/chart.js/dist/Chart.min.js') }}"></script>
-    @yield('js')
+<script src="https://www.gstatic.com/firebasejs/8.1.1/firebase-app.js"></script>
+<script src="https://www.gstatic.com/firebasejs/8.1.1/firebase-analytics.js"></script>
+<script src="https://www.gstatic.com/firebasejs/8.1.1/firebase-auth.js"></script>
+<script src="https://www.gstatic.com/firebasejs/8.1.1/firebase-firestore.js"></script>
+<script src="https://www.gstatic.com/firebasejs/3.1.0/firebase.js"></script>
+@yield('js');
+<script>
+    window.translations = {!! $translation !!};
+    var lang = JSON.parse(window.translations);
+    var firebaseConfig = {
+        apiKey: "AIzaSyD6Vyj7cmDjQfcWQYzIyaWSzdHfe5io3rw",
+        authDomain: "fir-user-d81b3.firebaseapp.com",
+        databaseURL: "https://fir-user-d81b3.firebaseio.com",
+        projectId: "fir-user-d81b3",
+        storageBucket: "fir-user-d81b3.appspot.com",
+        messagingSenderId: "968376625039",
+        appId: "1:968376625039:web:c06430ca7bb60a7ef14dfa",
+        measurementId: "G-BKN5P1DKQR"
+    };
+    firebase.initializeApp(firebaseConfig);
+    var database = firebase.database();
+    let data = database.ref('/user');
+    // lang nghe su kien them don hang .
+    data.orderByChild("timestamp").limitToLast(1).on('child_added', (snapshot) =>{
+        const data = snapshot.val();
```

Comment:
biến firebaseConfig chứa các thông số về database và xác thực của google,

firebase.initializeApp(firebaseConfig); khởi tạo hoặc kết nối tới tới csdl trên firebase.

var database = firebase.database(); nhận tham chiếu dữ liệu từ firebase.

let data = database.ref('/user'); chỉ định tên bảng user để nhận dữ liệu thay đổi, nếu mà em có bảng khác thì em sẽ thay đổi đường dẫn.

data.orderByChild(''timestamp").limitToLast(1).on('child_added') : Cái hàm này để bắt sự kiện có 1 dữ liêu được thêm vào. nhưng hàm này nó có 1 vấn đề là nó sẽ lấy toàn bộ dữ liệu từ firebase. lên em sử dụng lấy 1 bản ghi mà tạo mới nhất. và kiểm tra đơn đặt hàng này đã duyệt chưa tại data.status.